lib/reek/configuration/configuration_file_finder.rb

```diff
@@ -17,42 +17,102 @@ class ConfigFileException < StandardError; end
     # The order in which ConfigurationFileFinder tries to find such a
     # configuration file is exactly like above.
     module ConfigurationFileFinder
-      module_function
```

Comment:
Replaced module_function with class << self below because module_function + private is just a fucking mess in ruby. Seriously. Hopefully this gets fixed in Ruby 3.
The whole diff is a little messy unfortunately, might be easier to look at the whole file.